### PR TITLE
new testcases to v1/pod/util.go

### PR DESCRIPTION
**Pull request by tianshapjq**
**What this PR does / why we need it**:
some new testcases to v1/pod/util.go

### CODE REVIEW

**Comment by tianshapjq** [comment]
@nikhiljindal @yujuhong is it ok to my testcase?

**Comment by tianshapjq** [comment]
could this be /lgtm? @yujuhong

**Comment by tianshapjq** [comment]
@kubernetes/sig-testing-pr-reviews

File: pkg/api/v1/pod/util_test.go [hunk 1]
```diff
@@ -404,3 +405,140 @@ func TestIsPodAvailable(t *testing.T) {
...
+		{
+			status:   nil,
+			name:     "",
+			expected: ExpectedStruct{status: v1.ContainerStatus{}, exists: false},
+		},
+	}
+
+	for _, test := range tests {
+		resultStatus, exists := GetContainerStatus(test.status, test.name)
+		assert.Equal(t, test.expected.status, resultStatus, "unexpected test: %v, expected status: %v, actual: %v", test, test.expected.status, resultStatus)
```
Comment (yujuhong) [inline pkg/api/v1/pod/util_test.go]: No need to print the expected and resultStatus yourself. `assert` will do that if it fails.

File: pkg/api/v1/pod/util_test.go [hunk 2]
```diff
@@ -404,3 +405,140 @@ func TestIsPodAvailable(t *testing.T) {
...
+		assert.Equal(t, test.expected.exists, exists, "unexpected test: %v, expected exists: %v, actual: %v", test, test.expected.exists, exists)
```
Comment (yujuhong) [inline pkg/api/v1/pod/util_test.go]: Same as above.

File: pkg/api/v1/pod/util_test.go [hunk 3]
```diff
@@ -404,3 +405,140 @@ func TestIsPodAvailable(t *testing.T) {
...
+		exists bool
+	}
+
+	tests := []struct {
+		status   []v1.ContainerStatus
+		name     string
+		expected ExpectedStruct
+	}{
+		{
+			status:   []v1.ContainerStatus{{Name: "test1", Ready: false, Image: "image1"}},
```
Comment (yujuhong) [inline pkg/api/v1/pod/util_test.go]: Add another ContainerStatus in the test case to verify that the function retrieves the correct one.

File: pkg/api/v1/pod/util_test.go [hunk 4]
```diff
@@ -404,3 +405,140 @@ func TestIsPodAvailable(t *testing.T) {
...
+}
+
+func TestGetExistingContainerStatus(t *testing.T) {
+	tests := []struct {
+		status   []v1.ContainerStatus
+		name     string
+		expected v1.ContainerStatus
+	}{
+		{
+			status:   []v1.ContainerStatus{{Name: "test1", Ready: false, Image: "image1"}},
```
Comment (yujuhong) [inline pkg/api/v1/pod/util_test.go]: The test cases are the same as `TestGetContainerStatus`. Merging both into one?

File: pkg/api/v1/pod/util_test.go [hunk 5]
```diff
@@ -404,3 +405,140 @@ func TestIsPodAvailable(t *testing.T) {
...
+		{
+			status:   nil,
+			name:     "",
+			expected: v1.ContainerStatus{},
+		},
+	}
+
+	for _, test := range tests {
+		resultStatus := GetExistingContainerStatus(test.status, test.name)
+		assert.Equal(t, test.expected, resultStatus, "unexpected test: %v, expected status: %v, actual: %v", test, test.expected, resultStatus)
```
Comment (yujuhong) [inline pkg/api/v1/pod/util_test.go]: No need to print the expected and actual status yourself.

File: pkg/api/v1/pod/util_test.go [hunk 6]
```diff
@@ -404,3 +405,140 @@ func TestIsPodAvailable(t *testing.T) {
...
+				LastTransitionTime: metav1.NewTime(time.Add(1000))},
+			expected: true,
+		},
+	}
+
+	for _, test := range tests {
+		var resultStatus bool
+		resultStatus = UpdatePodCondition(test.status, &test.conditions)
+
+		assert.Equal(t, test.expected, resultStatus, "unexpected result of test %v, expected %v, actual %v", test, test.expected, resultStatus)
```
Comment (yujuhong) [inline pkg/api/v1/pod/util_test.go]: No need to print the expected and actual status yourself.

File: pkg/api/v1/pod/util_test.go [hunk 7]
```diff
@@ -404,3 +405,140 @@ func TestIsPodAvailable(t *testing.T) {
...
+func TestGetContainerStatus(t *testing.T) {
+	type ExpectedStruct struct {
+		status v1.ContainerStatus
+		exists bool
+	}
+
+	tests := []struct {
+		status   []v1.ContainerStatus
+		name     string
+		expected ExpectedStruct
```
Comment (yujuhong) [inline pkg/api/v1/pod/util_test.go]: Add a short description for each test case. Include this in your `assert` call later to clearly show which test case this is.

```
    desc string
```

```
   assert.Equal(t, test.expected.status, resultStatus, t.desc)
``` 

Please also do this for all the test functions below.

**Comment by tianshapjq** [comment]
@yujuhong thanks for the review. updates already committed, PTAL~

**Comment by tianshapjq** [comment]
@yujuhong hi, can this pr be merged?

**Comment by yujuhong** [comment]
> @yujuhong hi, can this pr be merged?

I don't have the permission to approve the PR.
/assign @thockin

**Pull request merged**